include/zephyr/zbus/zbus.h

```diff
@@ -348,19 +358,64 @@ struct zbus_channel_observation {
 	ZBUS_OBS_DECLARE(_observers);                                                     \
 	/* Create all channel observations from observers list */                         \
 	FOR_EACH_FIXED_ARG_NONEMPTY_TERM(_ZBUS_CHAN_OBSERVATION, (;), _name, _observers)
+
+/**
+ * @brief Zbus channel definition with numeric identifier.
+ *
+ * This macro defines a channel.
+ *
+ * @param _name The channel's name.
+ * @param _id The channel's unique numeric identifier.
+ * @param _type The Message type. It must be a struct or union.
+ * @param _validator The validator function.
+ * @param _user_data A pointer to the user data.
+ *
+ * @see struct zbus_channel
+ * @param _observers The observers list. The sequence indicates the priority of the observer. The
+ * first the highest priority.
+ * @param _init_val The message initialization.
+ */
+#define ZBUS_CHAN_ID_DEFINE(_name, _id, _type, _validator, _user_data, _observers, _init_val) \
+	static _type _CONCAT(_zbus_message_, _name) = _init_val;                          \
+	static struct zbus_channel_data _CONCAT(_zbus_chan_data_, _name) = {              \
+		.observers_start_idx = -1,                                                \
+		.observers_end_idx = -1,                                                  \
+		IF_ENABLED(CONFIG_ZBUS_PRIORITY_BOOST, (                                  \
+			.highest_observer_priority = ZBUS_MIN_THREAD_PRIORITY,            \
+		))                                                                        \
+	};                                                                                \
+	static K_MUTEX_DEFINE(_CONCAT(_zbus_mutex_, _name));                              \
+	_ZBUS_CPP_EXTERN const STRUCT_SECTION_ITERABLE(zbus_channel, _name) = {           \
+		ZBUS_CHANNEL_NAME_INIT(_name) /* Maybe removed */                         \
+		IF_ENABLED(CONFIG_ZBUS_CHANNEL_ID, (.id = _id,))                          \
+		.message = &_CONCAT(_zbus_message_, _name),                               \
+		.message_size = sizeof(_type),                                            \
+		.user_data = _user_data,                                                  \
+		.validator = _validator,                                                  \
+		.data = &_CONCAT(_zbus_chan_data_, _name),                                \
+		IF_ENABLED(ZBUS_MSG_SUBSCRIBER_NET_BUF_POOL_ISOLATION, (                  \
+			.msg_subscriber_pool = &_zbus_msg_subscribers_pool,               \
+		))                                                                        \
+	};                                                                                \
+	/* Extern declaration of observers */                                             \
+	ZBUS_OBS_DECLARE(_observers);                                                     \
+	/* Create all channel observations from observers list */                         \
+	FOR_EACH_FIXED_ARG_NONEMPTY_TERM(_ZBUS_CHAN_OBSERVATION, (;), _name, _observers)
+
```

Comment:
As mentioned in the PR description, I haven't found a way to do this properly as the `_observers` and `_init_val` parameters are actually lists, and adding the extra indirection expands the list into extra parameters.
I could resolve this by changing `, _observers, _init_val` to `...` and  expanding it in the common macro. This would make the macro args less clear though.